### PR TITLE
Fix #136 Removed temporary/instance variables that are not read and written, unnecessary assignments

### LINKED ISSUE #136: Fix Linting Issues
This is a large meta-issue that has many problems. The problems should be divided up.

- [x] Uses do: instead of collect: or select: (Weight: 4) @BenBals #138 
- [x] *Dot After Return Consistency* (Weight: 3) @robertkirsten #150 
      _(Please also check what the style guide says about last-line-dots without a return statement)_
- [x] ifTrue:/ifFalse returns instead of and:/or: (Weight: 1) @phlprcks -> was already refactored a while ago
- [x] intensive Coupling (Weight: 7) @sinnaj-r @BenBals #138
- [x] Law of Demeter (Weight: 9) @sinnaj-r @BenBals  #138
- [x] Long Methods (Weight: 4) @sinnaj-r #139
- [x] Sends questionable message (Weight: 1) @kej-jay 
- [ ] Sends add:/remove: to external collection (Weight: 1) @BenBals #146 
- [x] excessive number of arguments (Weight: 1) @phlprcks -> see comment below
- [x] Methods implemented but not sent (Weight: 1) @phlprcks #160 
- [x] Inconsistent method classification (Weight: 3, easy) @phlprcks -> see comment below
- [x] Temporary Variables not read and written (Weight: 3) @robertkirsten #153 
- [x] Instance Variable not read and written (Weight: 2) @robertkirsten #153 
- [x] Refers to class name instead of "self class" (Weight: 0.1) @kej-jay 
- [x] Unnecessary assignment to a temporary variable (Weight: 1) @robertkirsten #153
- [x] Unreferenced Variable (Weight: 3) @robertkirsten  #158 
	- The weight is so high, because the linter does not tell you which one (as far as I can tell)

| Team Member | Points |
|--|--|
| @sinnaj-r | 20 |
| @robertkirsten | 14 |
| @phlprcks | 6 |
| @kej-jay | 1.1 |
| @BenBals | 21 |

The weights are my subjective, imperfect measure of how time consuming and/or difficult a refactoring is as well as how much risk for escalation of time is involved. I suggest that each pair take on roughly equal amounts of weight. I welcome changes and more linting problems to be added.

This does not invalidate the "boy scout rule" that when you see one of these problems in the code you *must fix it immediatly*. That holds especially for simple refactorings like "Uses do: instead of collect: or select:".

### PR DESCRIPTION
**Pull request by robertkirsten**

### CODE REVIEW

File: packages/AlgernonTests.package/ALGTestCache.class/instance/testCacheKeyGenerationLevel1.st [hunk 1]
```diff
@@ -1,7 +1,5 @@
 testing
 testCacheKeyGenerationLevel1
-
-	| parent |
-	parent := ALGItem new.
+	ALGItem new.
```
Comment (robertkirsten) [inline packages/AlgernonTests.package/ALGTestCache.class/instance/testCacheKeyGenerationLevel1.st]: I am not sure whether this is necessary for the test.

File: packages/AlgernonTests.package/ALGTestAction.class/instance/setUp.st [hunk 2]
```diff
@@ -1,4 +1,3 @@
 running
 setUp
-	actions := ALGAction createListOfAlgeractionsGiven: (ALGClassType with: ALGCore).
-	self core: ALGCore new.
```
Comment (robertkirsten) [inline packages/AlgernonTests.package/ALGTestAction.class/instance/setUp.st:4]: I'm not 100% sure with this one, but from my testing removing this line doesn't alter the behavior. I modified a method in Algernon in such a way that the assertion fails, and it also did without this line. Plus removing this line causes the screen not to go white anymore when the tests in ALGTestAction are run.

File: packages/AlgernonTests.package/ALGTestKeyboard.class/instance/tearDown.st [hunk 3]
```diff
@@ -1,5 +1,3 @@
 running
 tearDown
-
-	super tearDown.
-	toolWindow ifNotNil: [ toolWindow delete ]
```
Comment (robertkirsten) [inline packages/AlgernonTests.package/ALGTestKeyboard.class/instance/tearDown.st:5]: Here too, the line seems to be unnecessary.

File: packages/AlgernonTests.package/ALGTestCache.class/instance/testCacheKeyGenerationLevel1.st (hunk 1 above)
Comment (BenBals) [reply packages/AlgernonTests.package/ALGTestCache.class/instance/testCacheKeyGenerationLevel1.st]: It should not be, a similar thing got removed here: https://github.com/hpi-swa-teaching/Algernon-Launcher/pull/155/files#diff-b1bbfaab62f80e23ef1bc09277ff9fd5

File: packages/AlgernonTests.package/ALGTestKeyboard.class/instance/tearDown.st (hunk 3 above)
Comment (BenBals) [reply packages/AlgernonTests.package/ALGTestKeyboard.class/instance/tearDown.st:5]: Yes, I think this is unnecessary. Please delete the method as that will be the same as calling `super tearDown`.

File: packages/AlgernonTests.package/ALGTestAction.class/instance/setUp.st (hunk 2 above)
Comment (BenBals) [reply packages/AlgernonTests.package/ALGTestAction.class/instance/setUp.st:4]: Seems good!

File: packages/AlgernonTests.package/ALGTestKeyboard.class/instance/tearDown.st (hunk 3 above)
Comment (BenBals) [reply packages/AlgernonTests.package/ALGTestKeyboard.class/instance/tearDown.st:5]: Jep, please remove the method all together and remove the instance variable.

**Comment by robertkirsten** [comment]
Do you have any idea why the CI run fails @BenBals?

**Comment by BenBals** [comment]
No, maybe @sinnaj-r has an idea.

**Comment by robertkirsten** [comment]
Works now.

File: .github/workflows/smalltalk-ci.yml [hunk 4]
```diff
@@ -23,4 +23,4 @@ jobs:
       shell: bash
       timeout-minutes: 15
       env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
```
Comment (sinnaj-r) [inline .github/workflows/smalltalk-ci.yml]: I have the feeling, that this change is not intended. (Also regarding to the not working CI)

Comment (robertkirsten) [reply .github/workflows/smalltalk-ci.yml]: This change is from a commit from @fniephaus which was added when I merged from development

Comment (robertkirsten) [reply .github/workflows/smalltalk-ci.yml]: Or is it? Wait I need to check

**Comment by robertkirsten** [comment]
So we are ready to merge?

File: README.md [hunk 5]
```diff
@@ -1,6 +1,4 @@
-# Algernon Launcher
```
Comment (sinnaj-r) [inline README.md]: This Looks like you are reverting changes here 🤔

**Review by sinnaj-r — APPROVED** [review]
Theres one strange readme change, but apart from that this can be merged 👍 
(After resolving the merge-conflicts)

**Review by BenBals — APPROVED** [review]
Lgtm

**Pull request merged**